python/ray/data/_internal/dataset_logger.py

```diff
@@ -71,5 +80,7 @@ def get_logger(self, log_to_stdout: bool = True):
         also removes the need for this getter method:
         `logger.info(msg="Hello world", stacklevel=2)`
         """
+        if not self._logger:
```

Comment:
Nit: Explicit `None` check rather than Falsey check.
https://peps.python.org/pep-0008/#programming-recommendations
https://google.github.io/styleguide/pyguide.html#2144-decision
```suggestion
        if self._logger is None:
```